### PR TITLE
Check if group IDs are valid before using them.

### PR DESCRIPTION
**Pull request by clokep**
I got annoyed at looking at these in Sentry all the time so I wanted to fix them once and for all, e.g. https://sentry.matrix.org/sentry/synapse-matrixorg/issues/127583/?query=is%3Aunresolved

### CODE REVIEW

File: synapse/rest/client/v2_alpha/groups.py [hunk 1]
```diff
@@ -25,6 +26,22 @@
 logger = logging.getLogger(__name__)
 
 
+def _validate_group_id(f):
+    """Wrapper to validate the form of the group ID.
+
+    Can be applied to any on_FOO methods that accepted a group ID as a URL parameter.
```
Comment (clokep) [inline synapse/rest/client/v2_alpha/groups.py:32]:
```suggestion
    Can be applied to any on_FOO methods that accepts a group ID as a URL parameter.
```

File: changelog.d/8977.bugfix [hunk 2]
```diff
@@ -0,0 +1 @@
+Properly return 400 errors on invalidate group IDs.
```
Comment (anoadragon453) [inline changelog.d/8977.bugfix:1]:
```suggestion
Properly return 400 errors on invalid group IDs.
```

File: synapse/rest/client/v2_alpha/groups.py [hunk 3]
```diff
@@ -25,6 +26,22 @@
...
+def _validate_group_id(f):
+    """Wrapper to validate the form of the group ID.
+
+    Can be applied to any on_FOO methods that accepts a group ID as a URL parameter.
+    """
+
+    @wraps(f)
+    def wrapper(self, request, group_id, *args, **kwargs):
+        if not GroupID.is_valid(group_id):
+            raise SynapseError(400, "%s was not legal group ID" % (group_id,))
```
Comment (anoadragon453) [inline synapse/rest/client/v2_alpha/groups.py:38]:
```suggestion
            raise SynapseError(400, "%s is not a legal group ID" % (group_id,))
```

I realise I'm changing an old error message here, but this seems more correct?

File: synapse/rest/client/v2_alpha/groups.py [hunk 4]
```diff
@@ -15,6 +15,7 @@
 # limitations under the License.
 
 import logging
+from functools import wraps
```
Comment (anoadragon453) [inline synapse/rest/client/v2_alpha/groups.py:18]: TIL about wraps vs decorators!

**Review by anoadragon453 — APPROVED** [review]
Couple tiny things, otherwise looks good! I'm glad all of our servlets follow the same `group_id` argument semantics :)

**Pull request merged**